mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/delivery/EventDelivery.java

```diff
@@ -48,5 +48,5 @@
     }
 
 
-    ExecutionStages deliver(Collection<MailboxListener> mailboxListeners, Event event);
+    ExecutionStages deliver(Collection<InVMEventBus.TypedRegistration> typedRegistrations, Event event);
```

Comment:
Why does the InVmDelivery depends on an object specific to the InVmEventBus?

Why is the InVMEventBus taking the decision of retrying or not? I feels like the decision is hidded from an EventBus perspective.

My proposal, which might be simpler to implement is to have 2 methods : `deliver` && `deliverWithRetries`